From 3fe9f1e131edaeba1ca4be124896d66a573ce61e Mon Sep 17 00:00:00 2001
From: Lukas Tribus <luky-37@hotmail.com>
Date: Sun, 19 May 2013 16:28:17 +0200
Subject: [PATCH 48/48] DOC: readme: add suggestion to link against static
 openssl

Adds a suggestion in README howto link against a static build of openssl.

This is useful if the OS includes an old openssl releas and recent features
or ciphers are required.
---
 README | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/README b/README
index 54ba1b7..9c0bca3 100644
--- a/README
+++ b/README
@@ -90,6 +90,17 @@ will automatically be linked with haproxy. Some systems also require libz, so
 if the build fails due to missing symbols such as deflateInit(), then try again
 with "ADDLIB=-lz".
 
+To link OpenSSL statically against haproxy, build OpenSSL with the no-shared
+keyword and install it to a local directory, so your system is not affected :
+
+    $ export STATICLIBSSL=/tmp/staticlibssl
+    $ ./config --prefix=$STATICLIBSSL no-shared
+    $ make && make install_sw
+
+When building haproxy, pass that path via ADDINC and ADDLIB to make and include
+libdl ("-ldl").
+    $ make TARGET=linux2628 USE_OPENSSL=1 ADDINC=-I$STATICLIBSSL/include ADDLIB="-L$STATICLIBSSL/lib -ldl"
+
 It is also possible to include native support for ZLIB to benefit from HTTP
 compression. For this, pass "USE_ZLIB=1" on the "make" command line and ensure
 that zlib is present on the system.
-- 
1.8.1.5

